packages/amplify-appsync-simulator/src/__tests__/server/subscription/websocket-server/server.test.ts

```diff
@@ -52,9 +52,10 @@ describe('WebsocketSubscriptionServer', () => {
   const onSubscribeHandler = jest.fn();
   const onConnectHandler = jest.fn();
   let serverPort = 20005;
-  const SUBSCRIPTION_PATH = '/graphql';
+  const SUBSCRIPTION_PATH = '/graphql/realtime';
```

Comment:
I got my brain wrapped around this for a bit. So, yes, since the localhost url won't match [the pattern](https://github.com/aws-amplify/amplify-js/blob/main/packages/pubsub/src/Providers/AWSAppSyncRealTimeProvider/index.ts#L69-L70) of an aws graphql endpoint, it falls back the a url pattern that has [realtime concatinated](https://github.com/aws-amplify/amplify-js/blob/main/packages/pubsub/src/Providers/AWSAppSyncRealTimeProvider/index.ts#L617-L620) onto it.

This seems like an artifact of the websocket cutover, supporting the beta endpoints along with release endpoints, given the special case logic I see around this. Is this the right pattern to maintain forward? Maybe it's not worth a change to all platforms for a better testing experience, but happy to have that question asked.

This behavior is released today, do customers have a workaround to enable them to use this?

Reply:
The handling of the API endpoints with custom domain (which is the case here with mock endpoint) is defined in this [Appsync doc](https://docs.aws.amazon.com/appsync/latest/devguide/real-time-websocket-client.html#handshake-details-to-establish-the-websocket-connection). I've verified that latest versions of Amplify Client libraries [JS](https://github.com/aws-amplify/amplify-js/pull/9171/files), [iOS](https://github.com/aws-amplify/aws-appsync-realtime-client-ios/pull/53/files#diff-af83dacdfaccd65d06c9374e7782b95ab7665aa32df46423b828b737ec9517c8) and [Android](https://github.com/aws-amplify/amplify-android/blob/0af580310b604fd297d25be77de668045594eecb/aws-api/src/main/java/com/amplifyframework/api/aws/SubscriptionEndpoint.java#L325) all respect this pattern and concatenate `realtime` to the path for custom domains. So I believe this is the right way going forward. The current status is that mock subscriptions do not work in any of the platforms unless the developer defines proxies in his app as described here. With this PR, we will fix that so that [such workaround is not needed](https://github.com/aws-amplify/amplify-category-api/issues/122#issuecomment-1129353449).